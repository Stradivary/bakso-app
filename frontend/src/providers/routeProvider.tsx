import { createBrowserRouter, RouterProvider } from "react-router-dom";
import { ProtectedRoute } from "../components/ProtectedRoute";
import { Register, MainPage, CustomerPage } from "../pages";

const router = createBrowserRouter([
  {
    path: '/auth',
    element: <Register />,
  },
  {
    path: '/',
    element: (
      <ProtectedRoute>
        <CustomerPage />
      </ProtectedRoute>
    ),
  },
  {
    path: '/maps',
    element: <CustomerPage />,
  },

]);


export function RouteProvider() {
  return (
    <RouterProvider router={router}  />
  );
}