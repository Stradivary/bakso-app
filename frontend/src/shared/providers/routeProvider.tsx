import { createBrowserRouter, RouterProvider } from "react-router-dom";
import { ProtectedRoute } from "../components/ProtectedRoute";
import { AuthPage, CustomerPage } from "../../pages";

const router = createBrowserRouter([
  {
    path: '/auth',
    element: <AuthPage />,
  },
  {
    path: '/',
    element: (
      <ProtectedRoute>
        <CustomerPage />
      </ProtectedRoute>
    ),
  },
]);


export function RouteProvider() {
  return (
    <RouterProvider router={router} />
  );
}