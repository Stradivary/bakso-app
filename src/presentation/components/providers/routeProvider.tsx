import { Container } from "@mantine/core";
import {
  createBrowserRouter,
  Outlet,
  RouteObject,
  RouterProvider,
} from "react-router-dom";
import { OfflineHandler } from "../OfflineHandler";
import { BroadcastMapPage } from "@/presentation/views/map";
import LoginView from "@/presentation/views/login/LoginView";
const routes: RouteObject[] = [
  {
    path: "/",
    element: (
      <>
        <Outlet />
        <OfflineHandler />
      </>
    ),
    children: [
      {
        index: true,
        element: <BroadcastMapPage />,
      },
      {
        path: "/login",
        element: <LoginView />,
      },
      {
        path: "*",
        element: (
          <Container h="100dvh">
            <h1>Oops, halaman tidak ditemukan</h1>

            <p>
              <a href="/">Kembali ke halaman utama</a>
            </p>
          </Container>
        ),
      },
    ] satisfies RouteObject[],
  },
] satisfies RouteObject[];

const router = createBrowserRouter(routes, {
  future: {
    v7_relativeSplatPath: true,
  },
});

export function RouteProvider() {
  return <RouterProvider router={router} />;
}
