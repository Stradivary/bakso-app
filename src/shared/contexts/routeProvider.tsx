import { Container } from "@mantine/core";
import {
  createBrowserRouter,
  Outlet,
  RouteObject,
  RouterProvider,
} from "react-router-dom";
import { OfflineHandler } from "@/views/components/OfflineHandler";
import { BroadcastMapPage } from "@/views/map";
import LoginView from "@/views/login/LoginView";
const routes: RouteObject[] = [
  {
    path: "/",
    element: (
      <>
        <Outlet />
        <OfflineHandler />
      </>
    ),
    children: [
      {
        index: true,
        element: <BroadcastMapPage />,
      },
      {
        path: "/login",
        element: <LoginView />,
      },
      {
        path: "*",
        element: (
          <Container h="100dvh">
            <h1>Oops, halaman tidak ditemukan</h1>

            <p>
              <a href="/">Kembali ke halaman utama</a>
            </p>
          </Container>
        ),
      },
    ] satisfies RouteObject[],
  },
] satisfies RouteObject[];

const router = createBrowserRouter(routes, {
  future: {
    v7_relativeSplatPath: true,
    v7_normalizeFormMethod: true,
    v7_skipActionErrorRevalidation: true,
    v7_fetcherPersist: true,
  },
});

export function RouteProvider() {
  return <RouterProvider router={router} />;
}
