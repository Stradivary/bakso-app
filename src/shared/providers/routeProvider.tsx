import { BroadcastMapPage } from "@/pages/map";
import LoginView from "@/shared/views/LoginView";
import { Container } from "@mantine/core";
import {
  createBrowserRouter,
  Outlet,
  RouteObject,
  RouterProvider,
} from "react-router-dom";
import { OfflineHandler } from "../components/OfflineHandler";
import { ThemeProvider } from "./themeProvider";

const routes: RouteObject[] = [
  {
    path: "/",
    element: (
      <ThemeProvider>
        <Outlet />
        <OfflineHandler />
      </ThemeProvider>
    ),
    children: [
      {
        index: true,
        element: <BroadcastMapPage />,
      },
      {
        path: "/login",
        element: <LoginView />,
      },
      {
        path: "*",
        element: (
          <Container h="100dvh">
            <h1>Oops, halaman tidak ditemukan</h1>

            <p>
              <a href="/">Kembali ke halaman utama</a>
            </p>
          </Container>
        ),
      },
    ] satisfies RouteObject[],
  },
] satisfies RouteObject[];

const router = createBrowserRouter(routes, {
  future: {
    v7_relativeSplatPath: true,
  },
});

export function RouteProvider() {
  return <RouterProvider router={router} />;
}
