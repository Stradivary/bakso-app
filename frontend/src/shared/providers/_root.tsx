import { MantineProvider } from "@mantine/core";
import { Notifications } from "@mantine/notifications";
import { AuthProvider } from "./authProvider";

import "@mantine/core/styles.css";
import "@mantine/notifications/styles.css";
import { RouteProvider } from "./routeProvider";


export function Root() {
  return (
    <AuthProvider>
      <MantineProvider theme={{ primaryColor: "red" }}>
        <RouteProvider />
        <Notifications />
      </MantineProvider>
    </AuthProvider>
  );
}

