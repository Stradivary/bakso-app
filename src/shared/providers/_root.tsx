
import "@mantine/core/styles.css";
import "@mantine/notifications/styles.css";
import "@fontsource/poppins/index.css";

import { AuthProvider } from "./authProvider";
import { RouteProvider } from "./routeProvider";
import { OfflineHandler } from "../components/OfflineHandler";
import { ThemeProvider } from "./themeProvider";
import { PropsWithChildren } from "react";

export function Contexts({ children }: PropsWithChildren) {
  return (
    <AuthProvider>
      <ThemeProvider>
        {children}
      </ThemeProvider>
    </AuthProvider>
  )
}

export function Root() {
  return (
    <Contexts>
      <RouteProvider />
      <OfflineHandler />
    </Contexts>
  );
}
