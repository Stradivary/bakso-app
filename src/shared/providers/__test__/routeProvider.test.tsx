import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { RouteProvider } from "../routeProvider";
import { Contexts } from "../_root";

describe("RouteProvider", () => {
  it("renders without crashing", () => {
    const { container } = render(
      <Contexts>

        <RouteProvider />
      </Contexts>
    );
    expect(container).toMatchSnapshot();
  });
});
