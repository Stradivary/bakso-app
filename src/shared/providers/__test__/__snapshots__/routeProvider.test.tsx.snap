// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RouteProvider > renders without crashing 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: 'Poppins', system-ui, sans-serif;--mantine-font-family-headings: 'Telkomsel Batik Sans', system-ui, sans-serif;--mantine-radius-default: calc(0.5rem * var(--mantine-scale));--mantine-primary-color-filled: var(--mantine-color-primary-filled);--mantine-primary-color-filled-hover: var(--mantine-color-primary-filled-hover);--mantine-primary-color-light: var(--mantine-color-primary-light);--mantine-primary-color-light-hover: var(--mantine-color-primary-light-hover);--mantine-primary-color-light-color: var(--mantine-color-primary-light-color);--mantine-primary-color-0: var(--mantine-color-primary-0);--mantine-primary-color-1: var(--mantine-color-primary-1);--mantine-primary-color-2: var(--mantine-color-primary-2);--mantine-primary-color-3: var(--mantine-color-primary-3);--mantine-primary-color-4: var(--mantine-color-primary-4);--mantine-primary-color-5: var(--mantine-color-primary-5);--mantine-primary-color-6: var(--mantine-color-primary-6);--mantine-primary-color-7: var(--mantine-color-primary-7);--mantine-primary-color-8: var(--mantine-color-primary-8);--mantine-primary-color-9: var(--mantine-color-primary-9);--mantine-color-primary-0: #ffe8ed;--mantine-color-primary-1: #ffcfd7;--mantine-color-primary-2: #fe9cab;--mantine-color-primary-3: #fd677d;--mantine-color-primary-4: #fc3a56;--mantine-color-primary-5: #fb203e;--mantine-color-primary-6: #fc1131;--mantine-color-primary-7: #e10224;--mantine-color-primary-8: #c9001f;--mantine-color-primary-9: #b00018;--mantine-color-secondary-0: #ebf4ff;--mantine-color-secondary-1: #d4e5fa;--mantine-color-secondary-2: #a3c9f7;--mantine-color-secondary-3: #71abf6;--mantine-color-secondary-4: #4b92f5;--mantine-color-secondary-5: #3782f5;--mantine-color-secondary-6: #2e7af6;--mantine-color-secondary-7: #2368dc;--mantine-color-secondary-8: #195dc5;--mantine-color-secondary-9: #0050ad;}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-primary-4);--mantine-color-primary-text: var(--mantine-color-primary-4);--mantine-color-primary-filled: var(--mantine-color-primary-8);--mantine-color-primary-filled-hover: var(--mantine-color-primary-9);--mantine-color-primary-light: rgba(252, 17, 49, 0.15);--mantine-color-primary-light-hover: rgba(252, 17, 49, 0.2);--mantine-color-primary-light-color: var(--mantine-color-primary-3);--mantine-color-primary-outline: var(--mantine-color-primary-4);--mantine-color-primary-outline-hover: rgba(252, 58, 86, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-4);--mantine-color-secondary-filled: var(--mantine-color-secondary-8);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-9);--mantine-color-secondary-light: rgba(46, 122, 246, 0.15);--mantine-color-secondary-light-hover: rgba(46, 122, 246, 0.2);--mantine-color-secondary-light-color: var(--mantine-color-secondary-3);--mantine-color-secondary-outline: var(--mantine-color-secondary-4);--mantine-color-secondary-outline-hover: rgba(75, 146, 245, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-primary-6);--mantine-color-primary-text: var(--mantine-color-primary-filled);--mantine-color-primary-filled: var(--mantine-color-primary-6);--mantine-color-primary-filled-hover: var(--mantine-color-primary-7);--mantine-color-primary-light: rgba(252, 17, 49, 0.1);--mantine-color-primary-light-hover: rgba(252, 17, 49, 0.12);--mantine-color-primary-light-color: var(--mantine-color-primary-6);--mantine-color-primary-outline: var(--mantine-color-primary-6);--mantine-color-primary-outline-hover: rgba(252, 17, 49, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-filled);--mantine-color-secondary-filled: var(--mantine-color-secondary-6);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-7);--mantine-color-secondary-light: rgba(46, 122, 246, 0.1);--mantine-color-secondary-light-hover: rgba(46, 122, 246, 0.12);--mantine-color-secondary-light-color: var(--mantine-color-secondary-6);--mantine-color-secondary-outline: var(--mantine-color-secondary-6);--mantine-color-secondary-outline-hover: rgba(46, 122, 246, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <h2>
    Unexpected Application Error!
  </h2>
  <h3
    style="font-style: italic;"
  >
    ResizeObserver is not defined
  </h3>
  <pre
    style="padding: 0.5rem; background-color: rgba(200, 200, 200, 0.5);"
  >
    ReferenceError: ResizeObserver is not defined
    at file:///workspaces/bakso-app/node_modules/@mantine/core/src/components/ScrollArea/use-resize-observer.ts:8:30
    at commitHookEffectListMount (/workspaces/bakso-app/node_modules/react-dom/cjs/react-dom.development.js:23189:26)
    at commitLayoutEffectOnFiber (/workspaces/bakso-app/node_modules/react-dom/cjs/react-dom.development.js:23312:15)
    at commitLayoutMountEffects_complete (/workspaces/bakso-app/node_modules/react-dom/cjs/react-dom.development.js:24727:9)
    at commitLayoutEffects_begin (/workspaces/bakso-app/node_modules/react-dom/cjs/react-dom.development.js:24713:7)
    at commitLayoutEffects (/workspaces/bakso-app/node_modules/react-dom/cjs/react-dom.development.js:24651:3)
    at commitRootImpl (/workspaces/bakso-app/node_modules/react-dom/cjs/react-dom.development.js:26862:5)
    at commitRoot (/workspaces/bakso-app/node_modules/react-dom/cjs/react-dom.development.js:26721:5)
    at performSyncWorkOnRoot (/workspaces/bakso-app/node_modules/react-dom/cjs/react-dom.development.js:26156:3)
    at flushSyncCallbacks (/workspaces/bakso-app/node_modules/react-dom/cjs/react-dom.development.js:12042:22)
  </pre>
  <p>
    ðŸ’¿ Hey developer ðŸ‘‹
  </p>
  <p>
    You can provide a way better UX than this when your app throws errors by providing your own 
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      ErrorBoundary
    </code>
     or
     
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      errorElement
    </code>
     prop on your route.
  </p>
</div>
`;
